pacbio-subreads-processing.wdl

```diff
@@ -24,172 +24,221 @@ import "structs.wdl" as structs
+        Int ccsChunks = 2
         Boolean generateFastq = false
         Map[String, String] dockerImages = {
             "bam2fastx": "quay.io/biocontainers/bam2fastx:1.3.0--he1c1bb9_8",
             "biowdl-input-converter": "quay.io/biocontainers/biowdl-input-converter:0.2.1--py_0",
-            "ccs": "quay.io/biocontainers/pbccs:4.2.0--1",
+            "ccs": "quay.io/biocontainers/pbccs:5.0.0--0",
             "fastqc": "quay.io/biocontainers/fastqc:0.11.9--0",
             "isoseq3": "quay.io/biocontainers/isoseq3:3.3.0--0",
             "lima": "quay.io/biocontainers/lima:1.11.0--0",
-            "multiqc": "quay.io/biocontainers/multiqc:1.9--pyh9f0ad1d_0"
+            "python3": "python:3.7-slim",
+            "multiqc": "quay.io/biocontainers/multiqc:1.9--pyh9f0ad1d_0",
+            "pacbio-merge": "lumc/pacbio-merge:0.2"
         }
     }
 
     meta {allowNestedInputs: true}
 
-    SubreadsConfig subreadsConfig = read_json(subreadsConfigFile)
-    Array[Subreads] allSubreads = subreadsConfig.subreads
+    # The name of the subreads, to be used to determine output filenames.
+    File subreadsName = basename(subreadsFile, ".subreads.bam")
+
+    # Get the CCS chunks.
+    call ccsChunks as createChunks {
+        input:
+            chunkCount = ccsChunks,
+            dockerImage = dockerImages["python3"]
+    }
+
+    # Index the input bamfile.
+    if (!defined(subreadsIndexFile)) {
+        call pbbam.Index as pbindex {
+            input:
+                bamFile = subreadsFile
+        }
+    }
+
+    File subreadsIndex = select_first([pbindex.index, subreadsIndexFile])
+    File subreadsBam = select_first([pbindex.indexedBam, subreadsFile])
+
+    scatter (chunk in createChunks.chunks) {
+        # Convert chunk from 1/10 to 1 to determine output filename.
+        String chunkNumber = sub(chunk, "/.*$", "")
 
-    scatter (subreads in allSubreads) {
         call ccs.CCS as ccs {
             input:
-                subreadsFile = subreads.subreads_file,
-                outputPrefix = outputDirectory + "/" + subreads.subreads_id + "/" + subreads.subreads_id,
+                subreadsFile = subreadsBam,
+                subreadsIndexFile = subreadsIndex,
+                outputPrefix = subreadsName + "_chunk" + chunkNumber,
                 cores = ccsCores,
+                chunkString = chunk,
                 dockerImage = dockerImages["ccs"]
         }
+    }
 
-        call lima.Lima as lima {
-            input:
-                libraryDesign = libraryDesign,
-                ccsMode = ccsMode,
-                splitBamNamed = splitBamNamed,
-                inputBamFile = ccs.ccsBam,
-                barcodeFile = subreads.barcodes_file,
-                outputPrefix = outputDirectory + "/" + subreads.subreads_id + "/" + subreads.subreads_id,
-                cores = limaCores,
-                dockerImage = dockerImages["lima"]
-        }
+    # Merge the bam files again.
+    call samtools.Merge as merge {
```

Comment:
Possibly provide the dockerImage here as well? To make sure the dockerImage can be changed through the `Map[String, String] dockerImages` input if necessary.

Reply:
I like it better when tasks just define their own docker image, otherwise you have to update in multiple places when you update to a more recent image. Users can still overwrite the docker image if they *really* want to via the workflow settings json.